packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema-v2.ts

```diff
@@ -340,7 +340,13 @@ const buildAPIProject = async (context: $TSContext, opts: TransformerProjectOpti
   checkForUnsupportedDirectives(schema, { dataSourceStrategies });
 
   const useBetaSqlLayer = context?.input?.options?.[USE_BETA_SQL_LAYER] ?? false;
-  const rdsLayerMapping = await getRDSLayerMapping(context, useBetaSqlLayer);
+
+  // Read the RDS Mapping S3 Manifest only if the schema contains SQL models or @sql directives.
+  let rdsLayerMapping: RDSLayerMapping | undefined = undefined;
+  if (containsSqlModelOrDirective(dataSourceStrategies)) {
```

Comment:
You need to pull the manifest if sqlDirectiveDataSourceStrategies is non-null also, since we might have a schema with SQL datasources limited to custom Queries & Mutations.

Reply:
Added additional check to verify sqlDirectiveDataSourceStrategies.